.sti/bin/assemble

```diff
@@ -1,52 +1,63 @@
 #!/bin/bash
 
-# For SCL enablement
-source .bashrc
+# This script uses these environment variables:
+#
+# DISABLE_COLLECTSTATIC: if not empty, inhibits execution of 'manage.py collectstatic'.
 
-set -e
+function assemble() {
+  # For SCL enablement
+  source .bashrc
 
-echo "---> Copying application source ..."
-cp -Rf /tmp/src/* ./
+  set -e
 
-if [ -f setup.py ]; then
-    echo "---> Installing application ..."
-    python setup.py install --user
-elif [ -f requirements.txt ]; then
-    echo "---> Installing dependencies ..."
-    pip install --user -r requirements.txt
-fi
+  echo "---> Copying application source ..."
+  cp -Rf /tmp/src/* ./
```

Comment:
I remember complaining about not being explicit, like it used to be when
that was introduced, sometime when we created the base image. I'm +1000 to
make it explicit back. @mfojtik gave me some reasoning behind that change
but I don't remember what was that.
On Jun 6, 2015 6:19 AM, "Ben Parees" <notifications@github.com> wrote:

> In .sti/bin/assemble
> <https://github.com/openshift/django-ex/pull/4#discussion_r31864417>:
>
> >
> > -if [ -f setup.py ]; then
> > -    echo "---> Installing application ..."
> > -    python setup.py install --user
> > -elif [ -f requirements.txt ]; then
> > -    echo "---> Installing dependencies ..."
> > -    pip install --user -r requirements.txt
> > -fi
> > +  echo "---> Copying application source ..."
> > +  cp -Rf /tmp/src/* ./
>
> i know you didn't add this and we do it in all our scripts, but as a rule
> copying to WORKDIR really bothers me. Seems like we ought to be explicit
> about where we're putting the app source. @mfojtik
> <https://github.com/mfojtik> @soltysh <https://github.com/soltysh> was
> there a reason we used this pattern that i'm forgetting?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/django-ex/pull/4/files#r31864417>.
>